pandas/tests/arrays/test_datetimelike.py

```diff
@@ -30,6 +67,41 @@ def test_astype_object(self, tz_naive_fixture):
         assert asobj.dtype == 'O'
         assert list(asobj) == list(dti)
 
+    @pytest.mark.parametrize('freqstr', ['D', 'B', 'W', 'M', 'Q', 'Y'])
+    def test_to_period(self, datetime_index, freqstr):
+        dti = datetime_index
+        arr = DatetimeArrayMixin(dti)
+
+        expected = dti.to_period(freq=freqstr)
+        result = arr.to_period(freq=freqstr)
+        assert isinstance(result, PeriodArrayMixin)
+
+        # placeholder until these become actual EA subclasses and we can use
+        #  an EA-specific tm.assert_ function
+        tm.assert_index_equal(pd.Index(result), pd.Index(expected))
```

Comment:
I am not sure we should test the equivalence between the index and array method in the array tests, that feels a bit out of place (since it is the index one that is calling the array)

Reply:
The point is that we have fairly robust tests for the Index methods.  Instead of effectively re-implementing all those tests, we can just test that the Index/Array methods behave identically.

Reply:
OK, that makes sense (at least for now) 
But you could still have fixtures for the arrays and create the index of it if needed inside the test (instead of currently the other way around) ?